applications/HDF5Application/custom_io/hdf5_file.cpp

```diff
@@ -4,12 +4,57 @@
 #include <sstream>
 #include <regex>
 #include <utility>
+#include <type_traits>
 #include "includes/kratos_parameters.h"
+#include "utilities/builtin_timer.h"
 
 namespace Kratos
 {
 namespace HDF5
 {
+namespace Internals
+{
+
+bool IsPath(const std::string& rPath)
+{
+#if defined(__GNUC__) && !defined(__clang__) && (__GNUC__ < 4 || (__GNUC__ == 4 && (__GNUC_MINOR__ < 9)))
+    KRATOS_ERROR << "This method is not compiled well. You should use a GCC 4.9 or higher" << std::endl;
+#else
+    return regex_match(rPath, std::regex("(/[\\w\\(\\)]+)+"));
+#endif
+}
+
+std::vector<std::string> Split(const std::string& rPath, char Delimiter)
```

Comment:
sth similar is also used in `model.cpp` => maybe this can be unified in the future

Reply:
Agreed, I don't know if that's the same, but Thomas also had something similar.